scripts/docker/borgserver

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -eux
+if [ -n "${SSH_KEY:-}" ]; then
+  dpkg-reconfigure openssh-server
+  sed -i \
+    -e 's/^#PasswordAuthentication yes$/PasswordAuthentication no/g' \
+    -e 's/^PermitRootLogin without-password$/PermitRootLogin no/g' \
+    /etc/ssh/sshd_config
+  if [[ $(ls /etc/ssh-keys/ssh_host_*) ]]; then
+    echo "Already initialized."
+  else
+    echo "Initializing ssh-keys folder..."
+    cp /etc/ssh/ssh_host_* /etc/ssh-keys
+  fi
+  chmod 400 /etc/ssh-keys/ssh_host_*
+  sed -i 's|/ssh/|/ssh-keys/|' /etc/ssh/sshd_config
+  sed -i '/^#HostKey/s/^#//' /etc/ssh/sshd_config
+  sed -e "s#SSH_KEY#${SSH_KEY}#g" /home/borg/authorized_keys.sample > /home/borg/.ssh/authorized_keys
```

Comment:
Hmm, where does this sample file come from? Debian I guess?
'
Okay, then it is empty.

Anyway, you should really suggest to save this file/dir in some (persistent) volume or prefill it (and not only `SSHFS_IDENTITY_FILE`), because otherwise you can get client connection errors…
Or oh well, this is the server part, nevermind I comment above…

Reply:
But should not this be configurable via a command-line, too? Otherwise, clients cannot connect if they are not authorized.

Also you'd like to restrict the commands to `borg serve` for security reasons.
See https://borgbackup.readthedocs.io/en/stable/deployment/central-backup-server.html#restrictions and https://borgbackup.readthedocs.io/en/stable/deployment/hosting-repositories.html

The script here could do that too/configure it properly…